testsuite/model/src/test/java/org/keycloak/testsuite/model/session/OfflineSessionPersistenceTest.java

```diff
@@ -122,6 +127,37 @@ public void testPersistenceSingleNodeDeleteRealm() {
         }
     }
 
+    @Test
+    @RequireProvider(value = HotRodConnectionProvider.class, only = DefaultHotRodConnectionProviderFactory.PROVIDER_ID)
+    public void testOfflineSessionsRemovedAfterDeleteRealm() {
+        String realmId2 = inComittedTransaction(session -> { return prepareRealm(session, "realm2").getId(); });
+        List<String> userIds2 = withRealm(realmId2, (session, realm) -> IntStream.range(0, USER_COUNT)
+                .mapToObj(i -> session.users().addUser(realm, "user2-" + i))
+                .map(UserModel::getId)
+                .collect(Collectors.toList())
+        );
+
+        try {
+            List<String> offlineSessionIds2 = createOfflineSessions(realmId2, userIds2);
+            assertOfflineSessionsExist(realmId2, offlineSessionIds2);
+
+            // Simulate server restart
+            reinitializeKeycloakSessionFactory();
+
+            assertOfflineSessionsExist(realmId2, offlineSessionIds2);
+
+            withRealm(realmId2, (session, realm) -> new RealmManager(session).removeRealm(realm));
```

Comment:
Shouldn't we use `session.reamls().removeRealm()` method here?

Reply:
Sure, changed.